### PR TITLE
Source Zendesk Support: fix missing headers

### LINKED ISSUE #23580: Source Zendesk Support: Issue with more than 100k organizations (Missing Headers)
## Environment
- **Airbyte version**:  example is 0.40.32
- **OS Version / Instance**: AWS EC2
- **Deployment**: Docker
- **Source Connector and version**: Zendesk-Support 0.2.25
- **Step where error happened**: Sync job

## Current Behavior
During the synchronization of the organization stream, I observe a random issue when querying pages above 1000 (Above 100k organizations, Zendesk has a different behaviour for pages above 1000 where you can query max 10 pages per minute and the response is different).

```
Traceback (most recent call last):
  File "/airbyte/integration_code/main.py", line 13, in <module>
    launch(source, sys.argv[1:])
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 131, in launch
    for message in source_entrypoint.run(parsed_args):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 122, in run
    for message in generator:
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 134, in read
    raise e
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 120, in read
    yield from self._read_stream(
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 189, in _read_stream
    for record in record_iterator:
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 255, in _read_incremental
    for message_counter, record_data_or_message in enumerate(records, start=1):
  File "/airbyte/integration_code/source_zendesk_support/streams.py", line 354, in read_records
    self._retry(request=request, retries=retries, original_exception=exc, **kwargs)
  File "/airbyte/integration_code/source_zendesk_support/streams.py", line 320, in _retry
    sleep_time = self.backoff_time(response)
  File "/airbyte/integration_code/source_zendesk_support/streams.py", line 132, in backoff_time
    retry_after = int(to_int(response.headers.get("Retry-After", 0)))
AttributeError: 'NoneType' object has no attribute 'headers'
```

I already fixed a similar issue in [PR #19967](https://github.com/airbytehq/airbyte/pull/19967) and it has come back here because it seems to not be a common issues as people using Airbyte do not seem to have more than 100k organizations.

The issue I had in this previous PR was the same : for this specific error, Zendesk doesn't fill a Header
```
Request URL: https://*******.zendesk.com/api/v2/organizations?start_time=978307200&page=1022, Response Code: 429, Response Text: Pagination limit reached for offset model
```

## Expected Behavior
I expect organizations to be correctly synchronized (it works for version Zendesk-Support 0.2.19).

## Are you willing to submit a PR?
Yes I am.

### PR DESCRIPTION
**Pull request by roman-yermilov-gl**
## What
- https://github.com/airbytehq/airbyte/issues/23580

## How
*Ensure `response` object exists before accessing its fields. Reorganize condition expression to take into account case when `response` is `None`*

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-zendesk-support test report (commit [`a93f468e27`](https://github.com/airbytehq/airbyte/pull/27765/commits/a93f468e27da43206257aab8171ebbab34da305b)) - ✅

⏲️  Total pipeline duration: 948 seconds

| Step                                                                          | Result   |
|:------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-zendesk-support/metadata.yaml | ✅        |
| Connector version semver check                                                | ✅        |
| Connector version increment check                                             | ✅        |
| QA checks                                                                     | ✅        |
| Code format checks                                                            | ✅        |
| Connector package install                                                     | ✅        |
| Build source-zendesk-support docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                    | ✅        |
| Acceptance tests                                                              | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-zendesk-support-fix-missing-headers/1687881732/a93f468e27da43206257aab8171ebbab34da305b/source-zendesk-support/0.8.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-zendesk-support test
```

**Comment by octavia-squidington-iii** [comment]
##  source-zendesk-support test report (commit [`a93f468e27`](https://github.com/airbytehq/airbyte/pull/27765/commits/a93f468e27da43206257aab8171ebbab34da305b)) - ✅

⏲️  Total pipeline duration: 934 seconds

| Step                                                                          | Result   |
|:------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-zendesk-support/metadata.yaml | ✅        |
| Connector version semver check                                                | ✅        |
| Connector version increment check                                             | ✅        |
| QA checks                                                                     | ✅        |
| Code format checks                                                            | ✅        |
| Connector package install                                                     | ✅        |
| Build source-zendesk-support docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                    | ✅        |
| Acceptance tests                                                              | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-zendesk-support-fix-missing-headers/1687886242/a93f468e27da43206257aab8171ebbab34da305b/source-zendesk-support/0.8.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-zendesk-support test
```

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (davydov-d) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:346]: Let's cover this case with a unit test please, because this is a second time we're doing this change according to the issue's reporter. If we had a unit test for this, we wouldn't have to fix it again and again

**Review by davydov-d — CHANGES_REQUESTED** [review]

**Review by lazebnyi — COMMENTED** [review]
LGTM! PR after @davydov-d requested changes will be approved.

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (roman-yermilov-gl) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:346]: Agree

Comment (roman-yermilov-gl) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:346]: Done

**Review by davydov-d — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-zendesk-support test report (commit [`d9b744ad8d`](https://github.com/airbytehq/airbyte/pull/27765/commits/d9b744ad8d2a30412bc7fad3dc268a27c060c6ec)) - ❌

⏲️  Total pipeline duration: 1356 seconds

| Step                                                                          | Result   |
|:------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-zendesk-support/metadata.yaml | ✅        |
| Connector version semver check                                                | ✅        |
| Connector version increment check                                             | ✅        |
| QA checks                                                                     | ✅        |
| Code format checks                                                            | ❌        |
| Connector package install                                                     | ✅        |
| Build source-zendesk-support docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                    | ✅        |
| Acceptance tests                                                              | ❌        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-zendesk-support-fix-missing-headers/1688129785/d9b744ad8d2a30412bc7fad3dc268a27c060c6ec/source-zendesk-support/0.8.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-zendesk-support test
```

**Comment by octavia-squidington-iii** [comment]
##  source-zendesk-support test report (commit [`67f23e5643`](https://github.com/airbytehq/airbyte/pull/27765/commits/67f23e5643aa9303ca2c3633cabec373f5b84958)) - ❌

⏲️  Total pipeline duration: 42.48s 

| Step                                                                          | Result   |
|:------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-zendesk-support/metadata.yaml | ✅        |
| Connector version semver check                                                | ✅        |
| Connector version increment check                                             | ✅        |
| QA checks                                                                     | ✅        |
| Code format checks                                                            | ✅        |
| Connector package install                                                     | ✅        |
| Build source-zendesk-support docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                    | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-zendesk-support-fix-missing-headers/1688140889/67f23e5643aa9303ca2c3633cabec373f5b84958/source-zendesk-support/0.8.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-zendesk-support test
```

**Comment by octavia-squidington-iii** [comment]
##  source-zendesk-support test report (commit [`67f23e5643`](https://github.com/airbytehq/airbyte/pull/27765/commits/67f23e5643aa9303ca2c3633cabec373f5b84958)) - ❌

⏲️  Total pipeline duration: 41.24s 

| Step                                                                          | Result   |
|:------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-zendesk-support/metadata.yaml | ✅        |
| Connector version semver check                                                | ✅        |
| Connector version increment check                                             | ✅        |
| QA checks                                                                     | ✅        |
| Code format checks                                                            | ✅        |
| Connector package install                                                     | ✅        |
| Build source-zendesk-support docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                    | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-zendesk-support-fix-missing-headers/1688141034/67f23e5643aa9303ca2c3633cabec373f5b84958/source-zendesk-support/0.8.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-zendesk-support test
```

**Comment by octavia-squidington-iii** [comment]
##  source-zendesk-support test report (commit [`67bd9f7a55`](https://github.com/airbytehq/airbyte/pull/27765/commits/67bd9f7a554ca35300adcfe39cfb4124d453e90b)) - ✅

⏲️  Total pipeline duration: 16mn21s 

| Step                                                                          | Result   |
|:------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-zendesk-support/metadata.yaml | ✅        |
| Connector version semver check                                                | ✅        |
| Connector version increment check                                             | ✅        |
| QA checks                                                                     | ✅        |
| Code format checks                                                            | ✅        |
| Connector package install                                                     | ✅        |
| Build source-zendesk-support docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                    | ✅        |
| Acceptance tests                                                              | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-zendesk-support-fix-missing-headers/1688143921/67bd9f7a554ca35300adcfe39cfb4124d453e90b/source-zendesk-support/0.8.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-zendesk-support test
```

**Pull request merged**
